govc/library/checkout.go

```diff
@@ -0,0 +1,141 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"context"
+	"flag"
+	"fmt"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/library"
+	"github.com/vmware/govmomi/vapi/library/finder"
+	"github.com/vmware/govmomi/vapi/rest"
+	"github.com/vmware/govmomi/vapi/vcenter"
+)
+
+type checkout struct {
+	*flags.ClusterFlag
+	*flags.FolderFlag
+	*flags.ResourcePoolFlag
+	*flags.HostSystemFlag
+}
+
+func init() {
+	cli.Register("library.checkout", &checkout{}, true)
+}
+
+func (cmd *checkout) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClusterFlag, ctx = flags.NewClusterFlag(ctx)
+	cmd.ClusterFlag.Register(ctx, f)
+
+	cmd.ResourcePoolFlag, ctx = flags.NewResourcePoolFlag(ctx)
+	cmd.ResourcePoolFlag.Register(ctx, f)
+
+	cmd.HostSystemFlag, ctx = flags.NewHostSystemFlag(ctx)
+	cmd.HostSystemFlag.Register(ctx, f)
+
+	cmd.FolderFlag, ctx = flags.NewFolderFlag(ctx)
+	cmd.FolderFlag.Register(ctx, f)
+}
+
+func (cmd *checkout) Process(ctx context.Context) error {
+	if err := cmd.ClusterFlag.Process(ctx); err != nil {
+		return err
+	}
+	if err := cmd.ResourcePoolFlag.Process(ctx); err != nil {
+		return err
+	}
+	if err := cmd.HostSystemFlag.Process(ctx); err != nil {
+		return err
+	}
+	return cmd.FolderFlag.Process(ctx)
+}
+
+func (cmd *checkout) Usage() string {
+	return "PATH NAME"
+}
+
+func (cmd *checkout) Description() string {
+	return `Check out Content Library item PATH to vm NAME.
+
+Examples:
+  govc library.checkout -cluster my-cluster my-content/template-vm-item my-vm`
+}
+
+func (cmd *checkout) Run(ctx context.Context, f *flag.FlagSet) error {
+	path := f.Arg(0)
+	name := f.Arg(1)
+
+	folder, err := cmd.FolderOrDefault("vm")
+	if err != nil {
+		return err
+	}
+	host, err := cmd.HostSystemIfSpecified()
+	if err != nil {
+		return err
+	}
+	cluster, err := cmd.ClusterIfSpecified()
+	if err != nil {
+		return err
+	}
+	pool, err := cmd.ResourcePoolIfSpecified()
+	if err != nil {
+		return err
+	}
+
+	return cmd.FolderFlag.WithRestClient(ctx, func(c *rest.Client) error {
+		m := library.NewManager(c)
+		res, err := finder.NewFinder(m).Find(ctx, path)
+		if err != nil {
+			return err
+		}
+		if len(res) != 1 {
+			return ErrMultiMatch{Type: "library", Key: "name", Val: path, Count: len(res)}
+		}
+		l, ok := res[0].GetResult().(library.Item)
+		if !ok {
+			return fmt.Errorf("%q is a %T", path, res[0].GetResult())
+		}
+
+		spec := vcenter.CheckOut{
+			Name: name,
+			Placement: &vcenter.Placement{
+				Folder: folder.Reference().Value,
+			},
+		}
+		if pool != nil {
+			spec.Placement.ResourcePool = pool.Reference().Value
+		}
+		if host != nil {
+			spec.Placement.Host = host.Reference().Value
+		}
+		if cluster != nil {
```

Comment:
Don't you require a cluster? Where would it go if Cluster isn't specified?

Reply:
One of `pool`, `host` or `cluster` is required for placement.  The error message from vCenter makes that clear if none are set.